modules/charon-core/src/main/java/org/wso2/charon3/core/config/SCIMSystemSchemaExtensionBuilder.java

```diff
@@ -0,0 +1,163 @@
+/*
+ * Copyright (c) 2025, WSO2 LLC. (http://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.charon3.core.config;
+
+import org.apache.commons.lang.StringUtils;
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.wso2.charon3.core.exceptions.CharonException;
+import org.wso2.charon3.core.exceptions.InternalErrorException;
+import org.wso2.charon3.core.schema.AttributeSchema;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Scanner;
+
+import static org.wso2.charon3.core.schema.SCIMConstants.SYSTEM_USER_SCHEMA_URI;
+
+/**
+ * This class is to build the extension system schema though the config file.
+ */
+public class SCIMSystemSchemaExtensionBuilder extends ExtensionBuilder {
+
+    private static final SCIMSystemSchemaExtensionBuilder instance = new SCIMSystemSchemaExtensionBuilder();
+    private static final Map<String, ExtensionAttributeSchemaConfig> extensionConfig = new HashMap<>();
+    private static final Map<String, AttributeSchema> attributeSchemas = new HashMap<>();
+    private AttributeSchema extensionSchema = null;
+    String extensionRootAttributeName = null;
+    String extensionRootAttributeURI;
```

Comment:
Does the default access modifier is given intentionally.

Reply:
Added proper access modifiers with dab02913610656de861ffb7a6360ae268ae16a95